core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -426,7 +426,7 @@ public TableMetadata removeSnapshotsIf(Predicate<Snapshot> removeIf) {
         // copy the log entries that are still valid
         newSnapshotLog.add(logEntry);
       } else {
-        // any invalid entry causes the history before it to be removed. otherwise, there could be
```

Comment:
oops, now that I read it the second time, the original one seems correct. Sorry my bad.